usaspending_api/common/api_request_utils.py

```diff
@@ -219,6 +219,33 @@ def range_intersect(self, fields, values):
         return Q(**q_case)
 
 
+# Handles unique value requests
+class UniqueValueHandler():
+    @staticmethod
+    # Data set to use (should be filtered already)
+    # Fields to find unique values for
+    # Returns a dictionary containing fields, values and their counts
+    # {
+    #   "recipient__name": {
+    #       "Jon": 5,
+    #       "Joe": 2
+    #   }
+    # }
+    def get_values_and_counts(data_set, fields):
+        data_set = data_set.all()  # Do this because we don't want to get finnicky with annotations
+        response_object = {}
+        if fields:
+            for field in fields:
```

Comment:
So I tried to use a method to annotate the data with the counts all at once, but it didn't function properly. What I had tried was: 
```data_set.value(field).annotate(count=Count(field))``` but it didn't seem to create the count as expect. If you have more experience with that aspect of Django I'd appreciate the help, as it should cut down on a few of the count calls. Couldn't find anything googling around.